### ISSUE TITLE
Add custom attributes to ES download of datasets

### ISSUE THREAD

**Issue by burkmarr** [BiologicalRecordsCentre/iRecord#932]
This new issue is branched from #800. Requires the ability to select a dataset for download and include all the custom attributes associated with it as the non-ES data download does.

**Comment by burkmarr** [comment]
A new simple simple `surveyFilter` has been created which allows ES records for reports and download to be quickly filtered on survey. Added a new `@columnsSurveyId` attribute on the `download` control which can be used to indicate that custom attributes are to be added to the download. The value of the attribute can be set to the ID of a survey or if specified and left unassigned and there is a `surveyFilter` control ont he page, the value is taken from any value selected there. When a download request is made, any custome sample and occurrence attributes are added to the CSV file. 

Pull requests to merge into develop branch: client_helpers (Indicia-Team/client_helpers#116), media (https://github.com/Indicia-Team/media/pull/42) and warehouse (Indicia-Team/warehouse#356).

**Comment by burkmarr** [comment]
@johnvanbreda - there is still this question about using es-filter-param: https://github.com/Indicia-Team/client_helpers/pull/116#pullrequestreview-504011683

**Comment by burkmarr** [comment]
@johnvanbreda - I've modified the code inline with your suggestions and updated the three pull requests on client_helpers, media and warehouse.

**Comment by johnvanbreda** [comment]
Comments added @burkmarr - thanks

**Comment by burkmarr** [comment]
Thanks @johnvanbreda. I've addressed the comments on the pull request. I have one outstanding issue on client_helpers - see question on pull request - so that isn't ready for merging yet.

**Comment by johnvanbreda** [comment]
@burkmarr all the pull requests are merged, so can this be closed?

**Comment by burkmarr** [comment]
Thanks @johnvanbreda. Yes I think that this can be closed now, but I'd rather leave it to @kitenetter so that he can review it before closing.

Martin, all the necessary code is merged into the develop branch which I've pulled onto iRecord dev so this can now be tried out there. I've added the control to the download sandbox - http://test.brc.ac.uk/irecord_dev/es-download-dev - with a bit of explanation about it.

It will be available for use on the live site when the develop branch is next merged into the master branch and deployed to the live site.
